apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c

```diff
@@ -0,0 +1,350 @@
+static void wifi_softap_sta_left(void);		// in softap mode, a station left
+static void wifi_sta_scan_ap_done(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res); // in station mode, scanning ap is done
+
+static wifi_manager_cb_s wifi_callbacks = {
+	NULL,	//wifi_sta_connected,
+	NULL,	//wifi_sta_disconnected,
+	NULL,	//wifi_softap_sta_joined,
+	NULL,	//wifi_softap_sta_left,
+	wifi_sta_scan_ap_done,	// in station mode, this callback function is called when scanning ap is done.
+};
+
+static wifi_manager_cb_s wifi_null_callbacks = {
+	NULL,	//wifi_sta_connected,
+	NULL,	//wifi_sta_disconnected,
+	NULL,	//wifi_softap_sta_joined,
+	NULL,	//wifi_softap_sta_left,
+	NULL,	// in station mode, this callback function is called when scanning ap is done.
+};
+
+static void wifi_sta_scan_ap_done(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res)
+{
+	printf("[pkes] T%d --> %s\n", getpid(), __FUNCTION__);
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (res == WIFI_SCAN_FAIL) {
+		printf("WiFi scan failed\n");
+		return;
+	}
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_info;
+	while (wifi_scan_iter != NULL) {
+		printf("SSID: %-20s, BSSID: %-20s, RSSI: %d, CH: %d, Phy_type: %d\n", \
+						wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi, \
+						wifi_scan_iter->channel, wifi_scan_iter->phy_mode);
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+}
+
+static void utc_wifi_manager_init_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(NULL);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_init_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_set_mode_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_set_mode(SOFTAP_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_FAIL);
+
+	wifi_manager_softap_config_s ap_config = { "PKES", 1, "1111122222" };
+
+	ret = wifi_manager_set_mode(WIFI_NONE, &ap_config);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_INVALID_ARGS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_set_mode_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	wifi_manager_softap_config_s ap_config = { "PKES", 1, "1111122222" };
+
+	ret = wifi_manager_set_mode(SOFTAP_MODE, &ap_config);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_get_mode_n(void)
```

Comment:
please change TC name and api name in TC macro to get_info

Reply:
I change these.